tests/src/Context/DrupalMailContext.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * @file
+ * Contains DrupalMailContext.
+ */
+
+namespace Drupal\nexteuropa\Context;
+
+use Behat\Behat\Hook\Scope\AfterScenarioScope;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use Behat\Gherkin\Node\TableNode;
+use function bovigo\assert\assert;
+use function bovigo\assert\predicate\equals;
+use Drupal\DrupalDriverManager;
+use Drupal\DrupalExtension\Context\DrupalSubContextInterface;
+use Drupal\DrupalExtension\Context\RawDrupalContext;
+
+/**
+ * Context for the internal mail handling.
+ *
+ * This context can be used for testing the mail features.
+ */
+class DrupalMailContext extends RawDrupalContext implements DrupalSubContextInterface {
+  /**
+   * The Drupal driver manager.
+   *
+   * @var DrupalDriverManager
+   */
+  protected $drupal;
+
+  /**
+   * Contains original variable values set during test execution.
+   *
+   * @var array
+   */
+  protected $variables = array();
+
+  /**
+   * The captured mail data.
+   *
+   * @var array
+   */
+  protected $capturedMail;
+
+  /**
+   * The variable context.
+   *
+   * @var VariableContext
+   */
+  protected $variableContext;
+
+  /**
+   * Constructs a DrupalMailContext object.
+   *
+   * @param DrupalDriverManager $drupal
+   *   The Drupal driver manager.
+   */
+  public function __construct(DrupalDriverManager $drupal) {
+    $this->drupal = $drupal;
+  }
+
+  /**
+   * Grabs other contexts we rely on from the environment.
+   *
+   * @BeforeScenario @internalMail
+   */
+  public function gatherContexts(BeforeScenarioScope $scope) {
+    $environment = $scope->getEnvironment();
+    $this->variableContext = $environment->getContext(VariableContext::class);
+  }
+
+  /**
+   * Clear the captured data.
+   *
+   * @AfterScenario @internalMail
+   */
+  public function clearCapturedData(AfterScenarioScope $scope) {
+    // Deleting the 'drupal_test_email_collector' variable.
+    variable_del('drupal_test_email_collector');
+  }
+
+  /**
+   * Configure Drupal to use the DrupalMailContext for mail handling.
+   *
+   * @Given platform is configured to use the internal mail handling
+   */
+  public function platformIsConfiguredToUseInternalMailHandling() {
+    $mail_system = array('default-system' => 'TestingMailSystem');
+    // Using VariableContext to set the specific variable. It will be
+    // restored to the previous value after finishing a scenario.
+    $this->variableContext->setVariable('mail_system', $mail_system);
+  }
+
+  /**
+   * Verifies that Drupal internal mail handler received the mail.
+   *
+   * @Then the internal mail handler received the mail
+   */
+  public function assertInternalMailHandlerReceivedTheMail() {
+    $captured_mail = self::getCapturedMail();
```

Comment:
I would use `$this->` instead of `self::` because `getCaptureMail()` manages non-static parameter; itself, it is not static.